### PR TITLE
GH-34428: [Python][Docs] Add docsstring for `make_fragment`

### LINKED ISSUE #34428: [Python] Add missing docs on `make_fragement`
### Describe the enhancement requested

I had to go to the code to see what we could pass into the `make_fragments`

### Component(s)

Python

### PR DESCRIPTION
**Pull request by Fokko**
* Closes: #34428

### CODE REVIEW

File: python/pyarrow/_dataset_parquet.pyx [hunk 1]
```diff
@@ -209,6 +209,26 @@ cdef class ParquetFileFormat(FileFormat):
...
+
+        Parameters
+        ----------
+        file : file-like object, path-like or str
+            The file or file path to make a fragment from.
+        filesystem : Filesystem, optional
+            If `filesystem` is given, `file` must be a string and specifies
+            the path of the file to read from the filesystem.
+        partition_expression : Expression, optional
+            The filter expression.
```
Comment (westonpace) [inline python/pyarrow/_dataset_parquet.pyx:223]:
```suggestion
            An expression that is guaranteed true for all rows in the fragment.  Allows
            fragment to be potentially skipped while scanning with a filter.
```

File: python/pyarrow/_dataset_parquet.pyx [hunk 2]
```diff
@@ -209,6 +209,26 @@ cdef class ParquetFileFormat(FileFormat):
...
+        row_groups : Iterable, optional
```
Comment (westonpace) [inline python/pyarrow/_dataset_parquet.pyx:224]: Iterable of what?  I assume numbers (I don't know if specify the iterable type elsewhere).  Also, maybe "The indices of the row groups to include"

Can this be empty?

**Review by westonpace — COMMENTED** [review]
Thanks for the continuing cleanup.

File: python/pyarrow/_dataset_parquet.pyx (hunk 1 above)
Comment (Fokko) [reply python/pyarrow/_dataset_parquet.pyx]: Thanks!

File: python/pyarrow/_dataset_parquet.pyx [hunk 3]
```diff
@@ -209,6 +209,26 @@ cdef class ParquetFileFormat(FileFormat):
...
+            The row groups to include
```
Comment (Fokko) [inline python/pyarrow/_dataset_parquet.pyx:226]:
```suggestion
            The indices of the row groups to include
```

File: python/pyarrow/_dataset_parquet.pyx (hunk 2 above)
Comment (Fokko) [reply python/pyarrow/_dataset_parquet.pyx:225]: Thanks @westonpace. I'd love to add types https://github.com/apache/arrow/issues/32609 but right now it is just guessing indeed.

I just checked, and an empty list `[]` is allowed, and will not ready any row groups, and return an empty table. Setting it to `None` will return all the row groups.

File: python/pyarrow/_dataset.pyx [hunk 4]
```diff
@@ -891,8 +891,13 @@ cdef class FileFormat(_Weakrefable):
         filesystem : Filesystem, optional
             If `filesystem` is given, `file` must be a string and specifies
             the path of the file to read from the filesystem.
-        partition_expression : Expression
+        partition_expression : Expression, optional
             The filter expression.
```
Comment (westonpace) [inline python/pyarrow/_dataset.pyx:895]:
```suggestion
            An expression that is guaranteed true for all rows in the fragment.  Allows
            fragment to be potentially skipped while scanning with a filter.
```

**Review by westonpace — APPROVED** [review]
Sorry, didn't realize there were two versions.  Same clarification on `partition_expression`.  Once that is in I think this is good to go.

File: python/pyarrow/_dataset.pyx (hunk 4 above)
Comment (Fokko) [reply python/pyarrow/_dataset.pyx]: Ah, I missed that one as well. Thanks!

**Comment by Fokko** [comment]
@westonpace Good to go!

**Pull request merged**